pkg/application/doguRestartUseCase.go

```diff
@@ -0,0 +1,82 @@
+package application
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type DoguRestartUseCase struct {
+	doguInstallationRepository doguInstallationRepository
+	blueprintSpecRepo          blueprintSpecRepository
+	restartRepository          domainservice.DoguRestartRepository
+}
+
+func NewDoguRestartUseCase(doguInstallationRepository doguInstallationRepository, blueprintSpecRepo blueprintSpecRepository, restartRepository doguRestartRepository) *DoguRestartUseCase {
+	return &DoguRestartUseCase{doguInstallationRepository: doguInstallationRepository, blueprintSpecRepo: blueprintSpecRepo, restartRepository: restartRepository}
+}
+
+func (useCase *DoguRestartUseCase) TriggerDoguRestarts(ctx context.Context, blueprintId string) error {
+	logger := log.FromContext(ctx).WithName("DoguRestartUseCase.TriggerDoguRestarts")
+	blueprintSpec, err := useCase.blueprintSpecRepo.GetById(ctx, blueprintId)
+	if err != nil {
+		return fmt.Errorf("could not get blueprint spec by id: %q", err)
+	}
+
+	logger.Info("searching for Dogus that need a restart...")
+	dogusThatNeedARestart := []common.SimpleDoguName{}
```

Comment:
🚫 **[checkstyle]** <ineffassign> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>ineffectual assignment to dogusThatNeedARestart